src/gluonts/dataset/repository/_m4.py

```diff
@@ -73,28 +75,29 @@ def generate_m4_dataset(
     # point available in pandas as the start date for each time series.
     mock_start_dataset = "1750-01-01 00:00:00"
 
-    save_to_file(
-        train_file,
-        [
-            to_dict(
-                target_values=target,
-                start=mock_start_dataset,
-                cat=[cat],
-                item_id=cat,
-            )
-            for cat, target in enumerate(train_target_values)
-        ],
-    )
+    train_data = [
+        to_dict(
+            target_values=target,
+            start=mock_start_dataset,
+            cat=[cat],
+            item_id=cat,
+        )
+        for cat, target in enumerate(train_target_values)
+    ]
+    test_data = [
+        to_dict(
```

Comment:
I think `to_dict` might not be needed since we do some conversion in the `DatasetWriter`. It might actually break the arrow dataset, since it assumes that you pass numpy arrays.